helix-core/src/syntax.rs

```diff
@@ -126,6 +129,22 @@ pub struct LanguageServerConfiguration {
     pub language_id: Option<String>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct FormatterConfiguration {
+    pub command: String,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub args: Vec<String>,
+    pub formatter_type: FormatterType,
```

Comment:
I tried that before and since `type` is a keyword I would have to use `r#type` which I think looks uglier.  If I were to use the `r#type`, would I access the field with `formatter.type` or would I still need r#?

Reply:
I guess since it's used once in the whole project we can afford to use `r#type`